great_expectations/rule_based_profiler/config/base.py

```diff
@@ -0,0 +1,215 @@
+import dataclasses
+from dataclasses import dataclass
+from typing import Any, Dict, List, Optional, Type
+
+from ruamel.yaml.comments import CommentedMap
+
+from great_expectations.data_context.types.base import BaseYamlConfig
+from great_expectations.marshmallow__shade import INCLUDE, Schema, fields, post_load
+from great_expectations.marshmallow__shade.decorators import post_dump
+from great_expectations.types import DictDot
+from great_expectations.util import filter_properties_dict
+
+
+class NotNullSchema(Schema):
+    """
+    Extension of Marshmallow Schema to facilitate implicit removal of null values before serialization.
+
+    The __config__ attribute is utilized to point a Schema to a configuration. It is the responsibility
+    of the child class to define its own __config__ to ensure proper serialization/deserialization.
```

Comment:
This class-level dunder var seems to align with Marshmallow conventions but let me know if it feels weird and we can adjust accordingly